src/components/discovery/DiscoverySearchForm.js

```diff
@@ -30,10 +42,13 @@ const CONDITION_RULES = [
             const isEnum = value.fieldSchema.hasOwnProperty("enum");
 
             // noinspection JSCheckFunctionSignatures
-            if (searchValue === null
-                    || (!isEnum && !searchValue)
-                    || (isEnum && !value.fieldSchema.enum.includes(searchValue))) {
-                cb("This field is required.");
+            if (
+                !VARIANT_OPTIONAL_FIELDS.includes(value.field) &&
+                (searchValue === null ||
```

Comment:
Mix of `searchValue === null` and `!searchValue` in condition chain, which can have different results (e.g. searchValue is undefined), is this intended? If not we could stick to `!searchValue`.

Reply:
everything below `!VARIANT_OPTIONAL_FIELDS.includes(value.field) &&` is legacy code, so I don't have an immediate answer. It's possible that some search values could be "0".